### DISCUSSION TITLE
Does the function to clear the form not work on FileUpload field?

### DISCUSSION THREAD

**Discussion by cesarlopes** [filamentphp/filament#8979] [Help]
Hello,

I tried using the function fill() to reset the form after submit, but only FileUpload was not cleared.

$this->form->fill();

**Comment by leandrocfe** [comment]
This was supposed to work. Could you share the code please?

**Comment by cesarlopes** [comment]
`
public function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('name'),
                FileUpload::make('trades_file')
                    ->label('Negociações B3')
                    ->preserveFilenames()
                    ->moveFiles()
                    ->storeFiles(false)
                ,
            ])->statePath('data');
    }

    protected function getFormActions(): array
    {
        return [
            Action::make('save')
                ->label('importar')
                ->submit('save'),
            Action::make('clear')
                ->action(function () {
                    $this->form->fill();
                })
        ];
    }

    public function save(): void
    {
        ....
    }
`

**Reply by leandrocfe** [reply]
any console error?

**Comment by sh4msi** [comment]
I have the same problem and no error is displayed in the console

**Comment by zeeshantariq08** [comment]
you can use set to clear up the data 
        ```
         Action::make('clear')
                ->action(function () {
                       $set('data.name', '',);
                    $set('data.trade_files', '',);
                })
```

**Reply by sh4msi** [reply]
Unfortunately, it didn't work, and the uploaded photo is still displayed.

**Comment by zeeshantariq08** [comment]
try this !
```
   protected function getFormActions(): array
      {
          return array_merge(parent::getFormActions(), [
              Action::make('clear')
                  ->action(function () {
                      $this->fillForm();
                  })
          ],
          );
      }
```
